opl/opl_sdl.c

```diff
@@ -173,8 +173,8 @@ static void FillBuffer(int16_t *buffer, unsigned int nsamples)
 
         for (i=0; i<nsamples; ++i)
         {
-            buffer[i * 2] = (int16_t) mix_buffer[i * 2];
-            buffer[i * 2 + 1] = (int16_t) mix_buffer[i * 2 + 1];
+            buffer[i * 2 + 1] = (int16_t) mix_buffer[i * 2];
```

Comment:
Actually in vanilla doom stereo is reversed too. In most sound cards OPL3 channel A connected to left speaker and channel B connected to right (in sound blaster for example). But there are some cards with reversed OPL3 connection. I think Paul Radek had card with reversed OPL3 when upgrading OPL code. I think we should make this configurable in i_oplmusic.c like this:

...
#define OPL_CH_A_BIT 0x10
#define OPL_CH_B_BIT 0x20
...
// Voices:
static unsigned int opl_leftchannel_bit = OPL_CH_A_BIT;
static unsigned int opl_rightchannel_bit = OPL_CH_B_BIT;
...

in SetChannelPan:
        if (pan >= 96)
        {
            reg_pan = opl_rightchannel_bit;
        }
        else if (pan <= 48)
        {
            reg_pan = opl_leftchannel_bit;
        }
        else
        {
            reg_pan = opl_leftchannel_bit | opl_rightchannel_bit;
        }
in I_OPL_InitMusic:
    if (opl_reversestereo)
    {
        opl_leftchannel_bit = OPL_CH_B_BIT;
        opl_rightchannel_bit = OPL_CH_A_BIT;
    }